spec/features/category_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.feature 'Categories', type: :feature do
+  let(:user) { create(:user) }
+
+  before do
+    login_as(user, scope: :user)
+  end
+
+  scenario 'User sees information on the root page' do
+    visit root_path
+
+    expect(page).to have_content(user.name)
+    expect(page).to have_content('Total cost for all categories')
+    expect(page).to have_link('Add a new category', href: new_category_path)
+  end
+
+  scenario 'User visits a category page' do
+    category = create(:category, user: user)
+    visit category_path(category)
+    expect(page).to have_content('Total cost for this category')
+  end
+end
```

Comment:
- [ ] Great work writing feature tests for the `Categories` component. As highlighted in the project requirements, you ought to write unit and integration tests for the major components of your application. Kindly write _**integration/feature**_ tests for the `Expenses` component that you created. The _**integration/feature**_ ensures that users' interactions with the application behave as expected. Kindly implement this change. 👍 💻

#### Project requirement

![image](https://github.com/mryeminthu/budget-app/assets/77106962/9dc668df-159f-4408-b948-4964d7b18297)